lazy-pages/src/sys.rs

```diff
@@ -134,7 +145,14 @@ pub unsafe fn user_signal_handler(info: ExceptionInfo) -> Result<(), Error> {
 
             let page_buf = PageBuf::new_from_vec(buffer_as_slice.to_vec())
                 .expect("Cannot panic here, because we create slice with PageBuf size");
```

Comment:
Can we remove this expect? We have to follow as safe as it could be style of code.

I want to make one more proposal: could we not to perform here conversion to vec and create page buf directly from slice? 
Such functionality should be added in core also:

Please, check the [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=bbb03e4b981be74ab8200c1d40be332f)

Isn't it more optimal solution? @grishasobol @ark0f @ukint-vs

Reply:
- Conversion from slice

From [`impl From<&[T]> for Box<[T]>`](https://doc.rust-lang.org/nightly/std/boxed/struct.Box.html#impl-From%3C%26%27_%20%5BT%5D%3E) docs:

> This conversion allocates on the heap and performs a copy of slice.

Inside it uses `RawVec::with_capacity` which 

> Creates a `RawVec` (on the system heap) with exactly the capacity and alignment requirements

- Conversion from `Vec`

From [`Vec::into_boxed_slice`] docs:

> Note that this will drop any excess capacity.

From [`impl From<Vec<T>> for Box<[T]>`](https://doc.rust-lang.org/nightly/std/boxed/struct.Box.html#impl-From%3CVec%3CT%2C%20A%3E%3E) (which calls [`Vec::into_boxed_slice`]) docs:

> If v has excess capacity, its items will be moved into a newly-allocated buffer with exactly the right capacity.

Documentation is misleading. Actually the second doc is the truth because [`Vec::into_boxed_slice`] uses [`Vec::shrink_to_fit`] inside.

- Conversion from slice to `Vec` (e.g. [`to_vec`])

It uses [`Vec::with_capacity`] inside which:

> This method is allowed to allocate for more elements than capacity

So, @breathx, it is more optimal solution for now but I should mention that these conversions have no guarantees. Even the lastest [`Vec::with_capacity`]'s quote exists only in beta and nightly (not in stable yet)

[`to_vec`]: https://doc.rust-lang.org/nightly/std/primitive.slice.html#method.to_vec
[`Vec::into_boxed_slice`]: https://doc.rust-lang.org/nightly/std/vec/struct.Vec.html#method.into_boxed_slice
[`Vec::shrink_to_fit`]: https://doc.rust-lang.org/nightly/std/vec/struct.Vec.html#method.shrink_to_fit
[`Vec::with_capacity`]: https://doc.rust-lang.org/nightly/std/vec/struct.Vec.html#method.with_capacity

Reply:
The only one possibility to throw error is different sizes. There is some misleading with to_vec yes, but as I can see in source code it creates with given slice size.